consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/ProposalBlockConsistencyValidator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.validation;
+
+import tech.pegasys.pantheon.consensus.ibft.ConsensusRoundIdentifier;
+import tech.pegasys.pantheon.consensus.ibft.IbftExtraData;
+import tech.pegasys.pantheon.consensus.ibft.payload.ProposalPayload;
+import tech.pegasys.pantheon.consensus.ibft.payload.SignedData;
+import tech.pegasys.pantheon.ethereum.core.Block;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ProposalBlockConsistencyValidator {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  public boolean validateProposalMatchesBlock(
+      final SignedData<ProposalPayload> signedPayload, final Block proposedBlock) {
+
+    if (!signedPayload.getPayload().getDigest().equals(proposedBlock.getHash())) {
+      LOG.debug("Invalid Proposal, embedded digest does not match block's hash.");
+      return false;
+    }
```

Comment:
Must validate Height of block matches Sequence.

Reply:
Validate that the block offered fits atop the chainhead, and its height matches our sequence number.

Reply:
done.